web/src/components/Inputs/SearchBar.tsx

```diff
@@ -0,0 +1,61 @@
+import { ChangeEvent } from "react";
+import { MdClose, MdSearch } from "react-icons/md";
+
+import IconButton from "./IconButton";
+import TextField from "./TextField";
+
+interface Props {
+  value: string;
+  setValue: (value: string) => void;
```

Comment:
I would say to pass this through as `onChange` and pass the `event`. This seems to imply the only thing it does is set the value when people might want to trigger fetches or various other stuff.

It's a touch more verbose for users, but more conformant with other input components.

Reply:
I tried to keep it simple by only updating the value. What do you mean by triggering fetches? Not familiar with that

Reply:
`setValue` is simple, but subtly implies you can't/shouldn't pass anything in except for a setter. I would just pass through the normal `onChange` API

Reply:
Oh I see what you were referring to. You can still use a handler in between but might as well just pass the event and make it clearer. I'll update it

Reply:
Oh wait, I now know why I chose to do that. So we currently have a clear functionality, and it is going to be hard to set the value to an empty string if we only have onChange with an event.

Reply:
I see, that makes sense. Maybe just change it to `onChange` then. We could try to make an event but seems like a lot of work for not a lot of gain. This is a nitpick anyway.